### PR TITLE
Use classic style config file assembly to add support for piboot

### PR DESCRIPTION
**Pull request by jawn-smith**
This PR adds piboot support to UC22.

It also modernizes the config file assembly of UC22 to match the classic server and desktop images.

### CODE REVIEW

**Comment by jawn-smith** [comment]
Here is the resulting config.txt:

```
[all]
kernel=kernel.img
cmdline=cmdline.txt
initramfs initrd.img followkernel

[pi4]
max_framebuffers=2
arm_boost=1

[all]
# Enable the audio output, I2C and SPI interfaces on the GPIO header. As these
# parameters related to the base device-tree they must appear *before* any
# other dtoverlay= specification
dtparam=audio=on
dtparam=i2c_arm=on
dtparam=spi=on

# Comment out the following line if the edges of the desktop appear outside
# the edges of your display
disable_overscan=1

# If you have issues with audio, you may try uncommenting the following line
# which forces the HDMI output into HDMI mode instead of DVI (which doesn't
# support audio output)
#hdmi_drive=2

[cm4]
# Enable the USB2 outputs on the IO board (assuming your CM4 is plugged into
# such a board)
dtoverlay=dwc2,dr_mode=host

[all]

# Enable the FKMS ("Fake" KMS) graphics overlay, and allocate 128Mb to the
# GPU memory
dtoverlay=vc4-fkms-v3d,cma-128

# Uncomment the following to enable the Raspberry Pi camera module firmware
#start_x=1
#gpu_mem=128

# Prior versions of Ubuntu Core customized the on-board LEDs so that the green
# ACT LED was a heartbeat, and the red PWR LED represented SD card activity.
# Uncomment the following lines if you wish to restore these customizations
#dtparam=act_led_trigger=heartbeat
#dtparam=pwr_led_trigger=mmc0

# Config settings specific to arm64
arm_64bit=1
dtoverlay=dwc2
```

File: gadget.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 volumes:
   pi:
     schema: mbr
```
Comment (anonymouse64) [inline gadget.yaml:3]: gentle nudge to see if this can be gpt in time for uc22 :-)

**Comment by alfonsosanchezbeato** [comment]
Thanks for this! I tried the branch, but I got some errors in the staging step:
```
+ snapcraftctl prime
Failed to copy '/home/ubuntu/src/snappy/pi-gadget/stage/cmdline.txt': no such file or directory.
Check the path and try again.
```

**Comment by jawn-smith** [comment]
@alfonsosanchezbeato Thanks for the testing! I had been using make rather than snapcraft in destructive mode, which was a mistake. It looks to be a simple path disagreement. `cmdline.txt` is in `stage/boot-assets/` so this should be an easy change to make.

File: gadget.yaml (hunk 1 above)
Comment (waveform80) [reply gadget.yaml:3]: Theoretically it's a trivial change, but I need to make sure it actually works on all (recent) firmware versions. There was a slightly disturbing changelog entry in the recent rpi-eeprom bump (https://bugs.launchpad.net/ubuntu/+source/rpi-eeprom/+bug/1965539) "FAT/GPT fixes and file-system performance improvements" so I want to either make sure that it all works (on SD, eMMC, and USB/SSD boot) and/or stick something in the release notes to the effect of "you must have firmware later than YYYYMMDD installed on your EEPROM to boot this image".

Unfortunately I haven't had the time to dig into this yet (it'll probably require a full day of flashing EEPROM versions and testing).

**Pull request merged**